BatteryStateManagement_proj.c

```diff
@@ -0,0 +1,83 @@
+/*****************************************************************************/
+/***    \file        BatteryStateManagement_proj.c
+***     \author      Naveen Balachandar (RBEI/ECK3) 
+***
+***     \brief       File to monitor the battery condition based required properties
+/*****************************************************************************/
+/*------ standard includes -------*/
+#include <stdio.h>
+#include <assert.h>
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    batteryCondMonitor_i
+ */
+/*!    \brief       predecting  the battery states
+ * 
+ *     \param       current temperature/soc/chargeRate ,respective min and max rate ,index 
+ *     \returns     batterystate -low/high/normal
+ *
+*//*------------------------------------------------------------------------*/
+int batteryCondMonitor_i(float batteryParameter ,float minRange, float maxRange,int batParIndex)
+{
+  /*Battery par printed for ref*/
+  char batPar[3][10] = {"temp","soc","chargerate"};
+  
+  if(batteryParameter < minRange)   /*Min range valid*/
+  {
+   printf("Battery parameter %s is low!\n",batPar[batParIndex]);
```

Comment:
Consider the situation when this message needs to be in different languages. The printf would become complex. Is there any way to make it more testable?